### PR TITLE
Python 3 compatibility for osg-test-log-viewer

### PR DESCRIPTION
**Pull request by matyasselmeci**
Tested with Python 2 and Python 3.4 on Fermicloud VMs, and Python 2.7 and 3.6 on my Fedora 28 laptop.

### CODE REVIEW

File: osg-test-log-viewer [hunk 1]
```diff
@@ -35,7 +37,8 @@ def load_logdata_from_handle(handle):
     current_test = ""
 
     for line in handle:
-        line = line.rstrip()
+        if six.PY3:
+            line = line.decode(errors="replace").rstrip()
```
Comment (edquist) [inline osg-test-log-viewer]: I presume `six.PY3` checks whether python3 is running - but what happens if this is false?  (Does `.rstrip()` still get run if it's python2?)

Comment (matyasselmeci) [reply osg-test-log-viewer]: Derp. Fixed.

Comment (edquist) [reply osg-test-log-viewer]: Yay, the system works! :D

**Comment by matyasselmeci** [comment]
We can probably drop Schrödinger's Python when we drop EL 7 support.

... in 2024.

**Comment by bbockelm** [comment]
Python3.4 available within EPEL6, no?  At least theoretically, we could drop python2 support for minor tools since python3 is on all our platforms.

**Comment by matyasselmeci** [comment]
I'd be okay with that; not sure how other people would feel.

There is some software we can't drop Python 2 support for yet because Python 3 versions of libraries they use aren't available on EL 6/EL 7 (I'm thinking osg-build and osg-configure, which use the RPM libraries).

**Comment by bbockelm** [comment]
For something that's purely OSG S&R facing such as `osg-test-log-viewer` (or `osg-test` in general!), I think it's an experiment worth doing.  Worst case, we annoy other developers.

I agree that we can sit back and wait on dependency chains to improve in general.  I would suspect that RHEL7 would eventually get python3 versions of the RPM bindings.

That said, `osg-build` and `osg-configure` should likely do `six` work like this.

@brianhlin ?

**Comment by edquist** [comment]
my quick thoughts - not really for or against -

- we can add a python3 requirement to osg-test to bring that in automatically (from epel if necessary)
- that might be annoying for unsuspecting users, but, for an internal tool, maybe that's not a problem
- installing osg-test from git does not automatically pull in the rpm requirements, including python3

**Comment by brianhlin** [comment]
I'm defintely on board for sixifying our internal tools. I'm not sure `osg-configure` falls under this category.

> installing osg-test from git does not automatically pull in the rpm requirements, including python3

We need to update the `bootstrap-osg-test` to pull in all the requirements but it's very low on the `osg-test` priority list.

**Comment by matyasselmeci** [comment]
BB is suggesting converting internal tools to Python 3 and sixifying external tools (the latter of which will probably be needed for EL 8 support).

**Pull request merged**